linux/gnome/src/main/java/com/github/weisj/darkmode/platform/linux/gnome/GnomeSettings.kt

```diff
@@ -0,0 +1,98 @@
+package com.github.weisj.darkmode.platform.linux.gnome
+
+import com.github.weisj.darkmode.platform.LibraryUtil
+import com.github.weisj.darkmode.platform.settings.*
+import com.google.auto.service.AutoService
+import kotlin.streams.toList
+
+/**
+ * Workaround for the fact that auto service currently doesn't work with singleton objects.
+ * https://github.com/google/auto/issues/785
+ */
+@AutoService(SettingsContainer::class)
+class GnomeSettingsProxy : SettingsContainer by GnomeSettings
+
+fun <T> concatenate(vararg lists: List<T>): List<T> {
+    val result: MutableList<T> = ArrayList()
+    lists.forEach { list: List<T> -> result.addAll(list) }
+    return result
+}
+
+object GnomeSettings : DefaultSettingsContainer() {
+    override val enabled: Boolean
+        get() = super.enabled && LibraryUtil.isGnome
+
+    /**
+     * This enum holds default values for the light, dark, and high contrast GTK themes.
+     * The defaults are a safe bet as they are included with almost any install of GTK.
+     * Even if they're not present, the way the logic works in GnomeThemeMonitorService,
+     * there won't be an issue and the plugin will fall back on the light theme.
+     *
+     * **See:** [Arch Wiki on GTK](https://wiki.archlinux.org/index.php/GTK#Themes)
+     */
+    private enum class DefaultGtkTheme(val info: GtkTheme) {
+        DARK(GtkTheme("Adwaita-dark")),
+        LIGHT(GtkTheme("Adwaita")),
+        HIGH_CONTRAST(GtkTheme("HighContrast")),
+    }
+
+    private const val DEFAULT_GUESS_LIGHT_AND_DARK_THEMES = true
+
+    @JvmField
+    var darkGtkTheme = DefaultGtkTheme.DARK.info
+
+    @JvmField
+    var lightGtkTheme = DefaultGtkTheme.LIGHT.info
+
+    @JvmField
+    var highContrastGtkTheme = DefaultGtkTheme.HIGH_CONTRAST.info
+
+    @JvmField
+    var guessLightAndDarkThemes = DEFAULT_GUESS_LIGHT_AND_DARK_THEMES
+
+    init {
+        if (!GnomeLibrary.get().isLoaded) {
+            throw IllegalStateException("Gnome library not loaded.")
+        }
+        group("Gnome Theme") {
+            val installedThemes = GnomeThemeUtils.getInstalledThemes()
+            /*
+             * The default themes are added to this list. They would already be added to the list because of their
+             * presence when initializing the `themes` vector in GnomeThemeUtils.cpp but because they are not
+             * the same instance as the defaults, the dropdown list would default to random themes because
+             * the instances of the three defaults couldn't be found in ChoiceProperty#choices.
+             */
+            val installedGtkThemes = concatenate(
```

Comment:
To avoid duplication of entries this could be
````kotlin
mutableSetOf(DefaultGtkTheme.DARK.info, DefaultGtkTheme.LIGHT.info, DefaultGtkTheme.HIGH_CONTRAST.info)
        .apply { addAll(installedThemes.map { GtkTheme(it) }) }
        .toList()
        .sorted()
````

Reply:
That's _much_ better. Thanks. It's sorted too, which is a nice touch!
4bee98d35c4534f74bee8d55afe9a2612021ca6f